subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -3160,12 +3182,20 @@ int lwm2m_perform_read_op(struct lwm2m_message *msg, uint16_t content_format)
 				ret = -EPERM;
 			} else {
 				/* start resource formatting */
-				engine_put_begin_r(&msg->out, &msg->path);
+				ret = engine_put_begin_r(&msg->out, &msg->path);
+				if (ret < 0) {
+					return ret;
+				}
 
 				/* perform read operation on this resource */
 				ret = lwm2m_read_handler(obj_inst, res,
 							 obj_field, msg);
-				if (ret < 0) {
+				if (ret == -ENOMEM) {
+					/* No point continuing if there's no
+					 * memory left in a message.
+					 */
+					return ret;
```

Comment:
Allowing to continue when `ENOMEM` is reported would require further rework fo the content writers - TLV or JSON will insert appropriate metadata in the packet first (TLV header, JSON tokens). If we fail to write the payload afterward, we still end up with a malformed packed (it should be required from content writers to undo any changes done to the packet). Therefore for now I've decided to treat `ENOMEM` as fatal.

Another thing is whether handling the requests partially is ok - responding with an error message could hint the server that it should query each resource individually, otherwise the event would go unnoticed, resulting in outdated reasource values on the server side.